src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -82,75 +84,34 @@ private async Task RunLoop(CancellationToken cancellationToken)
         {
             // Fetch them again, we just waited an hour
             AzureDevOpsTimelineOptions options = _options.Value;
-            
-            if (!int.TryParse(options.ParallelRequests, out int parallelRequests) || parallelRequests < 1)
-            {
-                parallelRequests = 5;
-            }
 
             if (!int.TryParse(options.BuildBatchSize, out int buildBatchSize) || buildBatchSize < 1)
             {
                 buildBatchSize = 1000;
             }
 
-            _logger.LogTrace(
-                "Opening connection to {organization} with {parallel} requests and access_token starting with '{token_sig}'",
-                options.AzureDevOpsOrganization,
-                parallelRequests,
-                options.AzureDevOpsAccessToken.Substring(0, 2));
-
-            var azureServer = new AzureDevOpsClient(
-                options.AzureDevOpsUrl,
-                options.AzureDevOpsOrganization,
-                parallelRequests,
-                options.AzureDevOpsAccessToken
-            );
-
-
-
             foreach (string project in options.AzureDevOpsProjects.Split(';'))
             {
-                await RunProject(azureServer, project, buildBatchSize, options, cancellationToken);
+                await RunProject(project, buildBatchSize, cancellationToken);
             }
         }
 
-        private async Task RunProject(
-            AzureDevOpsClient azureServer,
+        public async Task RunProject(
             string project,
             int buildBatchSize,
-            AzureDevOpsTimelineOptions options,
             CancellationToken cancellationToken)
         {
-
             DateTimeOffset latest;
-            try
+            DateTimeOffset? latestCandidate = await _timelineTelemetryRepository.GetLatestTimelineBuild(project);
+
+            if (latestCandidate == null)
```

Comment:
I generally prefer the ".HasValue" and ".Value", rather than the null check and the hard cast.  They are easier to see when I'm code reviewing what's going on.

Reply:
I had not considered this. I like it too.